lib/rules/var-spacing.rule.js

```diff
@@ -10,11 +10,77 @@
 
 'use strict';
 
+function validatePunctuators(current, next, context, node) {
+  next.forEach(function (nextPunctuator) {
+    if (current.loc.start.column !== nextPunctuator.loc.start.column) {
+      context.report({
+        message: 'Invalid indent!',
+        node:    node
+      });
+    }
+  });
+}
+
+function findNextPunctuators(node, source, lines) {
+  var after       = source.getTokenAfter(node),
+      nextLine    = lines[node.loc.start.line],
+      punctuators = [];
+
+  // in this case one big var assignment is declared:
+  //
+  // var var1 = {},
+  //     foo  = [];
+  // let blah = '';
+  if (node.declarations.length > 1) {
+    var nodeStartLine = node.loc.start.line;
+    punctuators = node.declarations.filter(function (subToken) {
+      return subToken.init !== null;
+    }).map(function (subToken) {
+      return source.getTokensBetween(subToken.id, subToken.init)[0];
+    }).filter(function (token, index) {
+      return token.type === 'Punctuator' && token.value === '=' && index > 0 && nodeStartLine < token.loc.start.line;
+    });
+  }
+
+  // check the next line for further expressions and skip completly in case of empty lines
+  if (typeof after !== 'undefined' && typeof nextLine !== 'undefined' && nextLine.trim().length !== 0) {
+    // checks whether a new variable is introduced
+    // or an existing one will be changed.
+    if (after.type === 'Keyword' && -1 !== ['var', 'let', 'const'].indexOf(after.value)) {
+      var expectedPunctuator = source.getTokenAfter(source.getTokenAfter(after));
+      if (expectedPunctuator.value === '=') {
+        punctuators.push(expectedPunctuator);
+      }
+    } else {
+      var identifier = after.type,
+          punctuator = source.getTokenAfter(after);
+
+      if (identifier === 'Identifier' && punctuator.type === 'Punctuator' && punctuator.value === '=') {
+        punctuators.push(punctuator);
+      }
+    }
```

Comment:
the above code snippet has two custom rules for new vars and overriding declared vars, but the logic looks quite duplicated in the end, so one approach should be introduced instead of the two